src/components/users/UserPoll.jsx

```diff
@@ -0,0 +1,184 @@
+import { instance } from "../../utils/axiosInstace";
+
+import { createTheme } from "@mui/material/styles";
+
+import { useDispatch, useSelector } from "react-redux";
+import {
+  getAllPolls,
+  getPollVoted,
+  votestarted,
+} from "../../redux/Slices/pollSlice";
+import { useNavigate } from "react-router-dom/dist";
+import { signout } from "../../redux/Slices/authSlice";
+import Alert from "@mui/material/Alert";
+import AlertTitle from "@mui/material/AlertTitle";
+import Stack from "@mui/material/Stack";
+import Loader from "../common/Loader";
+
+import Pagination from "@mui/material/Pagination";
+
+import { useEffect, useState } from "react";
+
+const theme = createTheme();
+
+theme.typography.h3 = {
+  fontSize: "1.2rem",
+  "@media (min-width:600px)": {
+    fontSize: "1.5rem",
+  },
+  [theme.breakpoints.up("md")]: {
+    fontSize: "2rem",
+  },
+};
+
+export default function UserPoll() {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const [isVoted, setVoted] = useState(false);
+  const [isLoading, setLoading] = useState(false);
+  const allListedPolls = useSelector((state) => state.poll.poll);
+  const voteLoader = useSelector((state) => state.poll.voteLoader);
+
+  const pollsperpage = 5;
+
+  const [page, setPage] = useState(1);
+
+  const handlePageChange = (event, value) => {
+    setPage(value);
+  };
+
+  const pollsToDisplay = allListedPolls.slice(
+    (page - 1) * pollsperpage,
+    page * pollsperpage
+  );
+
+  const totalPage = Math.ceil(allListedPolls.length / pollsperpage);
+
+  const handleLogout = () => {
+    localStorage.removeItem("polltoken");
+    dispatch(signout());
+    navigate("/");
+  };
+
+  const handleVotePollOption = (votePollId, optionText) => {
+    dispatch(votestarted({ votePollId: votePollId, optionText: optionText }));
+    const polltoken = localStorage.getItem("polltoken");
+    setTimeout(() => {
+      if (!isLoading) {
+        setVoted(false);
+      }
+    }, 2000);
+
+    instance
+      .get(`/do_vote?id=${votePollId}&option_text=${optionText}`, {
+        headers: {
+          access_token: polltoken,
+        },
+      })
```

Comment:
remove API call from every component and use redux-thunk which comes in build with toolkit